_includes/patterns-filter.html

```diff
@@ -347,52 +354,55 @@
   // Function called when user clicks on any of Technology checkboxes
   function showFilterCards() {
     let filterArray = [];
-
-    // Deselect the radio button when checkbox is selected
-    document.querySelectorAll("input[type=radio]:checked").forEach(function (el) {
-      el.checked = false;
-    });
-
     // Get all the checked checkboxes and add it to filterArray
     document
       .getElementById("compatibility-div")
       .querySelectorAll("input[type=checkbox]:checked")
       .forEach(function (el) {
         filterArray.push(el.value);
       });
-
-    // Select all pattern cards
-    // const patternCards = document.querySelectorAll(".patternCard");
-
-    // If no checkbox is selected, show all pattern cards and enable the "All" radio button
+      // Now filterArray has all the technologies checked by the user
+      // example: ["gcp", "keda"]
+      // we copy this filterArray to globalTechnologyList
+      globalTechnologyList = [...filterArray];
+      
+    // If no checkbox is selected, filter by category, by default category is set to "All"
     if (!filterArray.length) {
-      // for (let j = 0; j < patternCards.length; j++) {
-      //   patternCards[j].style.display = "block";
-      // }
-      document.getElementById("All").checked = true;
+      filterByType(globalCategorySelected);
+      return;
     } else {
 
-      // Loop over all pattern cards and check if any of the filter in filterArray is present in pattern card's filter attribute
-      // If yes, the show the card, else hide it
+      // Loop over all pattern cards and check if any of the filter in globalTechnologyList is present in pattern card's filter attribute
+      // If yes then set showCardTech to true
+      // also check if the selected category matches the card type (category the card belongs to)
+      // If yes then set showCardCat to true
+    
+      // If both showCardTech and showCardCat are true, then show the card otherwise hide it
```

Comment:
Thanks for making sure comments are updated

Reply:
Thank you :D